src/jit/_typeinfo.h

```diff
@@ -365,6 +367,13 @@ class typeInfo
         m_method = method;
     }
 
+    typeInfo(CORINFO_RESOLVED_TOKEN* token)
+    {
+        assert(token != nullptr && token->hMethod != nullptr && !isInvalidHandle(token->hMethod));
```

Comment:
This assert should be broken into three successive asserts for better diagnosability:
```
assert(token != nullptr);
assert(token->hMethod != nullptr);
assert(!IsInvalidHandle(token->hMethod));
```